src/tracing/types.rs

```diff
@@ -353,31 +373,44 @@ impl CallTraceNode {
                 to: self.trace.address,
                 value: self.trace.value,
                 gas: self.trace.gas_limit,
-                input: self.trace.data.clone(),
+                input: Bytes::new(),
+                // input: self.trace.data.clone(),
                 call_type: self.kind().into(),
                 // tx_type: self.trace.tx_type,
             }),
             CallKind::Create | CallKind::Create2 => Action::Create(CreateAction {
                 from: self.trace.caller,
                 value: self.trace.value,
                 gas: self.trace.gas_limit,
-                init: self.trace.data.clone(),
+                init: Bytes::new(),
+                // init: self.trace.data.clone(),
                 creation_method: self.kind().into(),
             }),
         }
     }
 
     /// Converts this call trace into an _empty_ geth [CallFrame]
     pub fn geth_empty_call_frame(&self, include_logs: bool) -> CallFrame {
+        self.geth_empty_call_frame_with_masking(include_logs, false)
+    }
+
+    /// Converts this call trace into an _empty_ geth [CallFrame] with optional data masking
+    pub fn geth_empty_call_frame_with_masking(&self, include_logs: bool, is_root_call: bool) -> CallFrame {
```

Comment:
same, `shield_outputs` instead of `is_root_call`

when you call it, the variable can be `is_root_call` so the reader can make the connection you allude to in the comments around here